src/linter/html/linter.ts

```diff
@@ -0,0 +1,43 @@
+import {taskStart} from "../../detectors/util/perf.js";
+import {extractScriptTags} from "../../detectors/transpilers/html/parser.js";
+import {LintMessageSeverity} from "../../detectors/AbstractDetector.js";
+import HtmlReporter from "./HtmlReporter.js";
+
+import type {TranspileResult} from "../../detectors/transpilers/AbstractTranspiler.js";
+import type {ReadStream} from "node:fs";
+
+export async function lintHtml(resourceName: string, contentStream: ReadStream): Promise<TranspileResult> {
+	const taskLintEnd = taskStart("Linting HTML", resourceName);
+	const report = new HtmlReporter(resourceName);
+
+	const scriptTags = await extractScriptTags(contentStream);
+	const jsScriptTags = scriptTags.filter((tag) => tag.attributes.every((attr) => {
```

Comment:
Why about doing this filtering already during parsing? Wouldn't this be more efficient?

Reply:
Done